runtime/parser2/expression.go

```diff
@@ -697,6 +698,25 @@ func defineDictionaryExpression() {
 	)
 }
 
+func defineIndexExpression() {
+	setExprLeftBindingPower(lexer.TokenBracketOpen, 150)
```

Comment:
This must be bumped to 160. It's actually a duplicate of the earlier definition here, which must be removed: https://github.com/onflow/cadence/blob/871304251b61fb152baac272bd1a084d6a0cd245/runtime/parser2/expression.go#L686

```suggestion
	setExprLeftBindingPower(lexer.TokenBracketOpen, 160)
```